src/goReferences.ts

```diff
@@ -10,73 +10,130 @@ import cp = require('child_process');
-			// get current word
-			let wordRange = document.getWordRangeAtPosition(position);
-			if (!wordRange) {
-				return resolve([]);
-			}
+	private runGoReference(document: vscode.TextDocument, position: vscode.Position, filename: string, cwd: string, options: { includeDeclaration: boolean }, resolve: any, reject: any): ChildProcess {
+		console.log("reference");
+		// get current word
+		let wordRange = document.getWordRangeAtPosition(position);
+		if (!wordRange) {
+			console.log("Can't find wordrange");
+			return null;
+		}
+		let textAtPosition = document.getText(wordRange);
+		let wordLength = textAtPosition.length;
+		let start = wordRange.start;
+		let possibleDot = '';
+		if (start.character > 0) {
+			possibleDot = document.getText(new vscode.Range(start.line, start.character - 1, start.line, start.character));
+		}
+		if (possibleDot === '.') {
+			let previousWordRange = document.getWordRangeAtPosition(new vscode.Position(start.line, start.character - 1));
+			let textAtPreviousPosition = document.getText(previousWordRange);
+			wordLength += textAtPreviousPosition.length + 1;
+		}
 
-			let goGuru = getBinPath('guru');
-			if (!path.isAbsolute(goGuru)) {
-				promptForMissingTool('guru');
-				return reject('Cannot find tool "guru" to find references.');
-			}
+		let offset = byteOffsetAt(document, position);
 
-			let filename = canonicalizeGOPATHPrefix(document.fileName);
-			let cwd = path.dirname(filename);
-			let offset = byteOffsetAt(document, position);
-			let env = getToolsEnvVars();
-			let buildTags = vscode.workspace.getConfiguration('go', document.uri)['buildTags'];
-			let args = buildTags ? ['-tags', buildTags] : [];
-			args.push('-modified', 'referrers', `${filename}:#${offset.toString()}`);
+		let gofindreferences = getBinPath('go-find-references');
+		console.log(['-file', filename, '-offset', offset.toString(), '-root', vscode.workspace.rootPath]);
+		let env = getToolsEnvVars();
+		return cp.execFile(gofindreferences, ['-file', filename, '-offset', offset.toString(), '-root', vscode.workspace.rootPath], { env }, (err, stdout, stderr) => {
+			try {
+				if (err && (<any>err).code === 'ENOENT') {
+					console.log("ENOENT");
+					vscode.window.showInformationMessage('The "go-find-references" command is not available.  Use "go get -v github.com/lukehoban/go-find-references" to install.');
+					return resolve(null);
+				}
+				let lines = stdout.toString().split('\n');
+				console.log(lines);
+				let results: vscode.Location[] = [];
+				for (let i = 0; i < lines.length; i += 2) {
+					let line = lines[i];
+					let match = /(.*):(\d+):(\d+)/.exec(lines[i]);
+					if (!match) continue;
+					let [_, file, lineStr, colStr] = match;
+					let referenceResource = vscode.Uri.file(path.resolve(cwd, file));
+					let range = new vscode.Range(
+						+lineStr - 1, +colStr - 1, +lineStr - 1, +colStr + wordLength - 1
+					);
+					results.push(new vscode.Location(referenceResource, range));
```

Comment:
for example here, would be a good candidate to do a ```const results = lines.map(() => {  put your logic in })```
ultimately same result, but it's better practice because you're following functional paradigms, which is the modern way to write Javascript.